docs/source/ext/yamltotable.py

```diff
@@ -0,0 +1,80 @@
+import os
+from docutils.parsers.rst.directives import tables
+import distributed
+from docutils import statemachine
+
+distributed_path = os.path.dirname(distributed.__file__)
+distributed_config = os.path.join(distributed_path, "distributed.yaml")
+from ruamel.yaml import YAML
+from ruamel.yaml.comments import CommentedMap
+
+yaml = YAML()
+yaml.preserve_quotes = True
+
+with open(distributed_config) as f:
+    _data = f.read()
+    yaml_conf = yaml.load(_data)
+
+
+class YamlToTable(tables.CSVTable):
+
+    required_arguments = 1
+
+    def __init__(self, *args, **kwargs):
+        # .. yamltotable:: scheduler argument
+        self.key = args[1][0]
+
+        super().__init__(*args, **kwargs)
+        self.data = yaml_conf["distributed"][self.key]
+
+    def get_csv_data(self):
+        return 1, self.state_machine.get_source(self.lineno - 1)
+
+    def parse_csv_data_into_rows(self, csv_data, dialect, source):
+        rows = []
+        for idx in self.data:
+            if not isinstance(self.data[idx], CommentedMap):
+                comment_token = self.data.ca.items.get(idx)
+                if comment_token is None:
+                    comment = ""
+                else:
+                    comment = comment_token[2].value
```

Comment:
Theoretically you could pull comments from another location besides the original yaml spec.  However, this may come back around to single-source of truth issues

Reply:
In dask-gateway @jcrist uses traitlets to store all of this. I'm not sure that Dask should go that far, but it's a demonstration of this approach.

This could also be useful for input validation